core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
+                    ,'template','alias','menutitle','link_attributes'
+                    ,'hidemenu','published','deleted'
+                    ,'parent','class_key','content_type','content_dispo'
+                )
+               ,'name'         => 'pagetitle'
+               ,'description'  => 'longtitle'
+               ,'content'      => 'content'
+               ,'icon'         => 'file-o'
+            );
+            /* user */
+            $elements['u'] = array(
+                'label'        => '(u)ser'
+               ,'type'         => 'users'
+               ,'class_key'    => 'modUser'
+               ,'permission'   => $this->modx->hasPermission('view_user')
+               ,'action'       => 'security/user/update'
+               ,'fields'       =>  array (
+                   'username','fullname','email','blocked'
+                  ,'username','fullname','dob','gender','photo'
+                  ,'email','phone','mobilephone','fax','website'
+                  ,'blocked','blockeduntil','blockedafter'    ,'failedlogincount','logincount','lastlogin','thislogin','sessionid'
+                  ,'address','country','city','state','zip'
+                  ,'comment','extended'
+                )
+               ,'name'         => 'username'
+               ,'description'  => 'email'
+               ,'content'      => 'extended'
+               ,'icon'         => 'user'
+            );
+            /*  system settings :x:ss:site_start: 
+                2do: Doesn't really make sense before we can open system settings for a specific key...
+            $elements['ss'] = array(
+                 'label'       => '(ss) systemSetting'
+                ,'type'        => ''
+                ,'class_key'   => 'modSystemSetting'
+                ,'permission'  => $this->modx->hasPermission('settings')
+                ,'action'      => 'system/settings'
+                ,'fields'      => array('key')
+                ,'name'        => 'key'
+                ,'description' => 'value'
+                ,'content'     => 'value'
+                ,'icon'        => 'gear'
+                
+            );
+            */
+            
+        }     
+        
+        
+        
+
+    /* Basic UI: Messages and validation */
+        $valid = false;
+        $step1 = $step2 = $step3 = $step4 = $step5 = false;
+    
+        /* What are we looking for? */
+        $start      = isset($query[0]) AND  empty($query) ? true : false;
+        $element    = isset($query[1]) ? $query[1] : false;
+        $field      = isset($query[2]) ? $query[2] : false;
+        $value      = isset($query[3]) ? $query[3] : false;
+        $page       = isset($query[4]) ? $query[4] : false;
+        
+        $count = count($query);
+        $result['count'] = $count;
+
+        $btn = "x-btn x-btn-small x-btn-icon-small-left primary-button x-btn-noicon";
+        
+        function validate_Uberbar($elements,$query){
```

Comment:
From a code style perspective, having a function inside a method feels very dirty. Maybe this can simply be moved to a protected/private method if you don't want it public?